thirdeye-ui/src/app/pages/welcome-onboard-datasource-select-datasets/welcome-onboard-datasource-select-datasets.component.tsx

```diff
@@ -0,0 +1,288 @@
+    } = useGetDatasets();
+
+    useEffect(() => {
+        getDatasets().then((datasetsProp) => {
+            if (!datasetsProp) {
+                return;
+            }
+            setSelectedDatasets(
+                Object.assign(
+                    {},
+                    ...datasetsProp?.map(({ name }) => ({ [name]: false }))
+                )
+            );
+        });
+    }, []);
+
+    useEffect(() => {
+        notifyIfErrors(
+            datasetsStatus,
+            errorMessages,
+            notify,
+            t("message.error-while-fetching", {
+                entity: t("label.datasets"),
+            })
+        );
+    }, [datasetsStatus]);
+
+    useEffect(() => {
+        notifyIfErrors(
+            datasourceStatus,
+            errorMessages,
+            notify,
+            capitalize(
+                t("message.error-while-fetching", {
+                    entity: t("label.datasources"),
+                })
+            )
+        );
+    }, [datasourceStatus]);
+
+    const areSomeChecked = Object.values(selectedDatasets).some((v) => v);
+    const areAllChecked = !Object.values(selectedDatasets).some((v) => !v);
+
+    const selectAllProps = useMemo(
+        () => ({
+            checked: areAllChecked,
+            indeterminate: areSomeChecked && !areAllChecked,
+            labelPrimaryText: "Select All",
+            name: SELECT_ALL,
+            onChange: handleToggleCheckbox,
+        }),
+        [areSomeChecked, areAllChecked]
+    );
+
+    const handleOnboardDatasets = useCallback(
+        (datasetsName: string[], datasourceName: string) =>
+            Promise.all(
+                datasetsName.map((datasetName) =>
+                    onBoardDataset(datasetName, datasourceName)
+                        .then(() => {
+                            notify(
+                                NotificationTypeV1.Success,
+                                t("message.onboard-success", {
+                                    entity: t("label.dataset"),
+                                })
+                            );
+                            // Redirect to welcome landing
+                            navigate(getWelcomeLandingPath());
+
+                            return Promise.resolve();
+                        })
+                        .catch((error: AxiosError) => {
+                            const errMessages = getErrorMessages(error);
+
+                            isEmpty(errMessages)
+                                ? notify(
+                                      NotificationTypeV1.Error,
+                                      t("message.onboard-error", {
+                                          entity: t("label.dataset"),
+                                      })
+                                  )
+                                : errMessages.map((err) =>
+                                      notify(NotificationTypeV1.Error, err)
+                                  );
```

Comment:
You could also use it here too:

```
 notifyIfErrors(
            ActionStatus.Error,
            getErrorMessages(error),
            notify,
            t("message.onboard-error", {
                entity: t("label.dataset"),
            })
        );
```

Reply:
Oops I missed this. Implemented it here as well.